File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02626/p02626_num8_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int N;
    cin >> N;

    vector<long long> A(N);
    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }

    long long sum = 0;
    long long maxElem = 0;
    for (int i = 1; i < N; i++) {
        sum += A[i];
        maxElem = max(maxElem, A[i]);
    }

    long long diff = abs(A[0] - sum);
    if (A[0] > sum && diff <= maxElem) {
        cout << diff << endl;
    } else {
        cout << -1 << endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02626/p02626_num8_parsed.py
import sys

def main():
    # Reading the input
    n = int(sys.stdin.readline())
    a = list(map(int, sys.stdin.readline().split()))

    # Calculating the total number of stones
    total_stones = sum(a)

    # Checking if there is only one pile
    if n == 1:
        print(-1)
        return

    # Checking if the sum of all piles is odd
    if total_stones % 2 != 0:
        print(-1)
        return

    # Checking if there is a pile with more stones than the sum of all other piles
    max_stones = max(a)
    if max_stones > total_stones - max_stones:
        print(-1)
        return

    # Calculating the minimum number of stones to move to guarantee Takahashi's win
    diff = (total_stones - max_stones) // 2
    if a[0] >= diff:
        print(1)
    else:
        print(-1)


if __name__ == '__main__':
    main()